src/plugins/data/common/search/aggs/metrics/avg_fn.ts

```diff
@@ -62,6 +62,13 @@ export const aggAvg = (): FunctionDefinition => ({
         defaultMessage: 'Represents a custom label for this aggregation',
       }),
     },
+    timeShift: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.metrics.timeShift.help', {
+        defaultMessage:
+          'Specifies whether the time range of documents used for the metric should be shifted by the specified amount',
```

Comment:
Can you make this a little more specific to canvas users? Something like `Shift the time range for the metric by a set time, for example 1h or 7d. "previous" will use the closest time range from the date histogram or time range filter.`